connector/src/test/java/it/eng/connector/integration/catalog/ArtifactIntegrationTest.java

```diff
@@ -0,0 +1,289 @@
+			.andExpect(status().isOk())
+			.andExpect(content().contentType(MediaType.APPLICATION_JSON))
+			.andReturn();
+		
+		String jsonList = resultList.getResponse().getContentAsString();
+		GenericApiResponse<List<Artifact>> apiRespList =  ToolsSerializer.deserializePlain(jsonList, typeRef);
+		  
+		assertNotNull(apiRespList.getData());
+		assertTrue(apiRespList.getData().size() > 2);
+		
+		MvcResult resultSingle = mockMvc.perform(
+    			get(ApiEndpoints.CATALOG_ARTIFACT_V1 + "/" + artifactFile.getId()).contentType(MediaType.APPLICATION_JSON))
+			.andExpect(status().isOk())
+			.andReturn();
+		
+		String jsonSingle = resultSingle.getResponse().getContentAsString();
+		GenericApiResponse<List<Artifact>> apiRespSingle = ToolsSerializer.deserializePlain(jsonSingle, typeRef);
+
+		assertNotNull(apiRespSingle);
+		assertTrue(apiRespSingle.isSuccess());
+		assertEquals(1, apiRespSingle.getData().size());
+		// Object equals won't work because the db Artifact has version and the static Artifact doesn't
+		assertEquals(artifactFile.getId(), apiRespSingle.getData().get(0).getId());
+		
+		//fail scenario
+		
+		MvcResult resultFail = mockMvc.perform(
+    			get(ApiEndpoints.CATALOG_ARTIFACT_V1 + "/" + "1").contentType(MediaType.APPLICATION_JSON))
+			.andExpect(status().isNotFound())
+			.andReturn();
+		
+		TypeReference<GenericApiResponse<Object>> typeRefFail = new TypeReference<GenericApiResponse<Object>>() {};
+
+		String jsonFail = resultFail.getResponse().getContentAsString();
+		GenericApiResponse<Object> apiRespFail = ToolsSerializer.deserializePlain(jsonFail, typeRefFail);
+		
+		assertNotNull(apiRespFail);
+		assertNull(apiRespFail.getData());
+		assertFalse(apiRespFail.isSuccess());
+		assertNotNull(apiRespFail.getMessage());
+		
+		// cleanup
+		artifactRepository.deleteById(artifactFile.getId());
+		artifactRepository.deleteById(artifactExternal.getId());
+	}
+	
+	@Test
+	@DisplayName("Artifact API - upload external")
+	@WithUserDetails(TestUtil.API_USER)
+	public void uploadArtifactExternal() throws Exception {
+		Dataset dataset = Dataset.Builder.newInstance()
+				.hasPolicy(Set.of(CatalogMockObjectUtil.OFFER))
+				.build();
+		datasetRepository.save(dataset);
+		
+		TypeReference<GenericApiResponse<Artifact>> typeRef = new TypeReference<GenericApiResponse<Artifact>>() {};
+		
+		MockPart part = new MockPart("url", CatalogMockObjectUtil.ARTIFACT_EXTERNAL.getValue().getBytes());
+		
+		MvcResult result = mockMvc.perform(
+				multipart(ApiEndpoints.CATALOG_ARTIFACT_V1 + "/upload/" + dataset.getId())
+				.part(part))
+			.andExpect(status().isOk())
+			.andExpect(content().contentType(MediaType.APPLICATION_JSON))
+			.andReturn();
+		
+		String json = result.getResponse().getContentAsString();
+		GenericApiResponse<Artifact> apiResp =  ToolsSerializer.deserializePlain(json, typeRef);
+		  
+		assertTrue(apiResp.isSuccess());
+		assertEquals(CatalogMockObjectUtil.ARTIFACT_EXTERNAL.getValue(), apiResp.getData().getValue());
```

Comment:
Can we somehow check, even query datasetRepository or artifactRepository, if data is actually present in Mongo?